src/test/ui/nll/mir_check_cast_unsize.stderr

```diff
@@ -5,17 +5,13 @@ LL |     x
    |     ^
 
 error: unsatisfied lifetime constraints
-  --> $DIR/mir_check_cast_unsize.rs:17:46
+  --> $DIR/mir_check_cast_unsize.rs:19:5
    |
-LL |   fn bar<'a>(x: &'a u32) -> &'static dyn Debug {
-   |  ________--____________________________________^
-   | |        |
-   | |        lifetime `'a` defined here
-LL | |     //~^ ERROR unsatisfied lifetime constraints
-LL | |     x
-LL | |     //~^ WARNING not reporting region error due to nll
-LL | | }
-   | |_^ return requires that `'a` must outlive `'static`
+LL | fn bar<'a>(x: &'a u32) -> &'static dyn Debug {
+   |        -- lifetime `'a` defined here
+LL |     //~^ ERROR unsatisfied lifetime constraints
+LL |     x
+   |     ^ cast requires that `'a` must outlive `'static`
```

Comment:
in this case I sort of prefer the error =)

Reply:
but I suppose it's mostly the crazy span; I think we are labeling the wrong point

Reply:
"cast requires..." is not a great label, but the spans are correct in my mind, although I would try and detect this kind of case and point at the return type:

```
error: unsatisfied lifetime constraints
  --> $DIR/mir_check_cast_unsize.rs:19:5
   |
LL | fn bar<'a>(x: &'a u32) -> &'static dyn Debug {
   |        --     ---         -------- `x` must be `'static` due to this return type
   |        |      |
   |        |      `x` is `'a` due to this type definition
   |        lifetime `'a` defined here
LL |     x
   |     ^ this is `'a` but it must outlive `'static`
```